protocol/kafkaapi/api_versions_response.go

```diff
@@ -0,0 +1,147 @@
+package kafkaapi
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/kafka-tester/protocol/decoder"
+	"github.com/codecrafters-io/kafka-tester/protocol/errors"
+	"github.com/codecrafters-io/kafka-tester/protocol/kafkaapi/headers"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+type ApiVersionsResponse struct {
+	Header headers.ResponseHeader
+	Body   ApiVersionsResponseBody
+}
+
+func (r *ApiVersionsResponse) Decode(response []byte, logger *logger.Logger) (err error) {
+	decoder := decoder.NewDecoder(response, logger)
+
+	decoder.BeginSubSection("ApiVersions Response")
+
+	defer func() {
+		decoder.EndCurrentSubSection()
+
+		if decodingErr, ok := err.(*errors.PacketDecodingError); ok {
+			detailedError := decodingErr.AddContexts("ApiVersions Response")
```

Comment:
I think a lot of complexity for this comes from the ordering - `AddContexts` is only done when the error is constructed, instead of before the decoding is done. Might be worth looking into merging the ideas of "SubSection" and "Context" (again, can be separate PR)